sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/implementation/ServiceBusReceiveLinkProcessorTest.java

```diff
@@ -253,11 +254,15 @@ void newLinkOnClose() {
         when(link2.getCredits()).thenReturn(1);
         when(link3.getCredits()).thenReturn(1);
 
+        when(link1.closeAsync()).thenReturn(Mono.empty());
+        when(link2.closeAsync()).thenReturn(Mono.empty());
+        when(link3.closeAsync()).thenReturn(Mono.empty());
+
         final ServiceBusReceiveLink[] connections = new ServiceBusReceiveLink[]{link1, link2, link3};
         final ServiceBusReceiveLinkProcessor processor = createSink(connections).subscribeWith(linkProcessor);
 
         // Act & Assert
-        StepVerifier.create(processor)
+        StepVerifier.create(processor.take(count))
```

Comment:
If only 4 messages are added to the `sink`, why is `take(count)` necessary?

Reply:
It specifically complete the reactor stream after count instead of "ask for next and not receiving and completing".